app/src/main/java/com/example/makeitso/screens/sign_up/SignUpViewModel.kt

```diff
@@ -73,33 +73,17 @@ class SignUpViewModel @Inject constructor(
             return
         }
 
-        viewModelScope.launch(showErrorExceptionHandler) {
-            val oldUserId = accountService.getUserId()
-            accountService.createAccount(email, password) { error ->
-                if (error == null) {
-                    linkWithEmail()
-                    updateUserId(oldUserId, openAndPopUp)
-                } else onError(error)
-            }
+        launchCatching {
+            val oldUserId = accountService.currentUserId
+            accountService.createAccount(email, password)
+            linkWithEmail()
```

Comment:
After linking the new account, we need to update the userId in the documents by calling `updateUserId(oldUserId)`. This method still does not exist in this class, but basically it should be like this:

```
private fun updateUserId(oldUserId: String) {
        launchCatching {
            val newUserId = accountService.currentUserId
            storageService.updateUserId(oldUserId, newUserId)
        }
    }
```

Reply:
Done